### PR TITLE
SYNCT-278 Fetch real objects and send merged object

### PR DESCRIPTION
**Pull request by wojtekbu**
* Added fetching end sending merged object
* Added handling merged conflict sync
*Added redirection on success

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/sync2/api/service/impl/SyncPushServiceImpl.java [hunk 1]
```diff
@@ -92,6 +92,37 @@ public AuditMessage readAndPushObjectToParent(String category, Map<String, Strin
         return auditMessage;
     }
 
+    @Override
+    public AuditMessage mergeForcePush(Object merged, String category, Map<String, String> resourceLinks,
```
[the remark this answers is not in the document]
Comment (wojtekbu) [reply api/src/main/java/org/openmrs/module/sync2/api/service/impl/SyncPushServiceImpl.java]: Audit model is not ready to represent complex relationship, so I've decided that conflict resolution will have connection only with conflict result audit message

File: api/src/main/java/org/openmrs/module/sync2/api/service/impl/SyncPushServiceImpl.java [hunk 2]
```diff
@@ -92,6 +92,37 @@ public AuditMessage readAndPushObjectToParent(String category, Map<String, Strin
...
+        } else {
+            auditMessage.setDetails(parentAudit.getDetails());
+            auditMessage.setSuccess(parentAudit.getSuccess());
+        }
+
+        if (parentAudit.getSuccess() && childAudit.getSuccess()) {
+            auditMessage = successfulMessage(auditMessage);
+        } else {
+            auditMessage.setDetails(childAudit.getDetails());
+            auditMessage.setSuccess(childAudit.getSuccess());
```
[the remark this answers is not in the document]
Comment (wojtekbu) [reply api/src/main/java/org/openmrs/module/sync2/api/service/impl/SyncPushServiceImpl.java]: ok

File: api/src/main/java/org/openmrs/module/sync2/api/service/impl/SyncPushServiceImpl.java [hunk 3]
```diff
@@ -92,6 +92,37 @@ public AuditMessage readAndPushObjectToParent(String category, Map<String, Strin
...
+            }
+        } else {
+            auditMessage.setDetails(parentAudit.getDetails());
+            auditMessage.setSuccess(parentAudit.getSuccess());
+        }
+
+        if (parentAudit.getSuccess() && childAudit.getSuccess()) {
+            auditMessage = successfulMessage(auditMessage);
+        } else {
+            auditMessage.setDetails(childAudit.getDetails());
```
[the remark this answers is not in the document]
Comment (wojtekbu) [reply api/src/main/java/org/openmrs/module/sync2/api/service/impl/SyncPushServiceImpl.java]: ok

File: omod/src/main/java/org/openmrs/module/sync2/web/controller/rest/SyncConflictRestController.java [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseBody;
+
+import java.io.IOException;
+
+import static org.openmrs.module.sync2.api.utils.SyncUtils.extractUUIDFromResourceLinks;
+
+@Controller("sync2.SyncConflictRestController")
+@RequestMapping(value = "/rest/sync2/conflict", produces = MediaType.APPLICATION_JSON_VALUE)
+public class SyncConflictRestController {
```
[the remark this answers is not in the document]
Comment (areklalo) [reply omod/src/main/java/org/openmrs/module/sync2/web/controller/rest/SyncConflictRestController.java]: I am not sure but if the ConflictResolutionController is page controller then regarding OpenMRS convention I believe should have the "Page" word in the name and should be in the correct directory.

File: api/src/main/java/org/openmrs/module/sync2/api/service/impl/SyncPushServiceImpl.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (areklalo) [reply api/src/main/java/org/openmrs/module/sync2/api/service/impl/SyncPushServiceImpl.java]: Ok so please update the ticket description.

**Pull request merged**